File: content/snippets/cpp/language_basics_strings.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

int main() {
    std::string first = "hello";
    std::string second = "world";
    std::string joined = first + " " + second;

    std::string part = joined.substr(0, 5);
    std::string replaced = joined;
    replaced.replace(0, 5, "hi");

    std::vector<std::string> words;
    std::istringstream iss(joined);
    for (std::string token; iss >> token;) {
        words.push_back(token);
    }

    std::cout << "joined: " << joined << "\n";
    std::cout << "part: " << part << "\n";
    std::cout << "replaced: " << replaced << "\n";
    std::cout << "tokens: ";
    for (const auto& w : words) {
        std::cout << w << ' ';
    }
    std::cout << "\n";
    return 0;
}

File: content/snippets/python/language_basics_strings.py
#!/usr/bin/env python3

def main() -> None:
    first = "hello"
    second = "world"
    joined = first + " " + second

    part = joined[:5]
    replaced = joined.replace("hello", "hi", 1)
    words = joined.split()

    print("joined:", joined)
    print("part:", part)
    print("replaced:", replaced)
    print("tokens:", words)


if __name__ == "__main__":
    main()
